Source/ModelGroups/Compatibility.cs

```diff
@@ -1,219 +1,200 @@
             }
-        }
-    }
 
-    // Used to add a property to the root level, or into an existing property
-    public class ExperimentalGltf1 : glTFLoader.Schema.Gltf
-    {
-        public ExperimentalGltf1() { }
-        public ExperimentalGltf1(glTFLoader.Schema.Gltf parent)
-        {
-            foreach (PropertyInfo property in parent.GetType().GetProperties())
+            void SetModelShouldLoad(List<Property> properties, string loadableStatus = ":white_check_mark:")
             {
-                var parentProperty = property.GetValue(parent);
-                if (parentProperty != null)
-                {
-                    property.SetValue(this, parentProperty);
-                }
+                properties.Add(new Property(PropertyName.ModelShouldLoad, loadableStatus));
             }
+
+            this.Models = new List<Model>
+            {
+                CreateModel((properties, gltf) => {
+                    SetVersionCurrent(properties, gltf);
+                    SetModelShouldLoad(properties);
+                }),
+                CreateModel((properties, gltf) => {
+                    SetVersionFuture(properties, gltf);
+                    SetDescription(properties, gltf, "Light object added at root");
+                    SetModelShouldLoad(properties);
+                }),
+                CreateModel((properties, gltf) => {
+                    SetVersionFuture(properties, gltf);
+                    SetDescription(properties, gltf, "Light property added to node object");
+                    SetModelShouldLoad(properties);
+                }),
+                CreateModel((properties, gltf) => {
+                    SetVersionFuture(properties, gltf);
+                    SetDescription(properties, gltf, "Alpha mode updated with a new enum value, and a fallback value");
+                    SetModelShouldLoad(properties);
+                }),
+                CreateModel((properties, gltf) => {
+                    SetMinVersion(properties, gltf);
+                    SetVersionFuture(properties, gltf);
+                    SetDescription(properties, gltf, "Requires a specific version or higher");
+                    SetModelShouldLoad(properties, "Only in version 2.1 or higher");
+                }),
+                CreateModel((properties, gltf) => {
+                    SetVersionCurrent(properties, gltf);
+                    SetDescription(properties, gltf, "Extension required");
+                    SetModelShouldLoad(properties, ":x:");
+                }),
+            };
+
+            GenerateUsedPropertiesList();
         }
 
-        // Creates a new root level property
-        public Light lights { get; set; }
-        public class Light
+        /// <summary>
+        /// These properties are added to the model after the Runtime layer generates the gltf.
+        /// </summary>
+        public override void PostRuntimeChanges(List<Property> combo, ref glTFLoader.Schema.Gltf gltf)
```

Comment:
This is still weird. I think a better solution is to attach a PostRuntimeChanges function to the model class and then you can set that to whatever you want to change for that model. Then just call model.PostRuntimeChanges when creating the schema version.

Reply:
Isn't that exactly what we're doing already?

Reply:
No. Currently, you have a PostRuntimeChanges function on ModelGroup and using the readme property to decide what to do. It will be much cleaner and clearer if each model can have a function associated with it that includes what runtime changes are necessary per model.

Reply:
Oh! I see. For some reason I understood 'Model Class' to mean the model group that the model was created by, instead of the literal model class. My bad, I think I've got it now.

Reply:
I have this mostly done, but I'm having an issue with setting the action. In the following code `postRuntimeChanges` isn't set to the desired action because it isn't a complex object and thus isn't passed by reference. Do you know of a solution to this other than making it into an object?

```
                // Sets up a function to apply properties to the model after the Runtime layer generates the gltf.
                Action<glTFLoader.Schema.Gltf> postRuntimeChanges = null;

                // Apply the properties that are specific to this gltf.
                setProperties(properties, gltf, postRuntimeChanges);

                // Create the gltf object
                return new Model
                {
                    Properties = properties,
                    GLTF = gltf,
                    PostRuntimeChanges = postRuntimeChanges,
                };
```

Then below
```
            this.Models = new List<Model>
            {
                CreateModel((properties, gltf, postRuntimeChanges) => {
                    SetVersionFuture(properties, gltf);
                    SetDescription(properties, "Light object added at root");
                    SetModelShouldLoad(properties);
                    postRuntimeChanges = (glTFLoader.Schema.Gltf schemaGltf) => { SetPostRuntimeAtRoot(schemaGltf); };
                }),
            }
```

Reply:
```
Model CreateModel(Action<List<Property>, Runtime.GLTF> setProperties, Action<glTFLoader.Schema.Gltf> postRuntimeChanges)
{
    // ...
    return new Model
    {
        // ...
        PostRuntimeChanges = postRuntimeChanges
    }
}

Reply:
Fixed. The code works as is, but I feels like what I just added could still be improved. At least the formatting of this section

```
            this.Models = new List<Model>
            {
                CreateModel((properties, gltf, PostRuntimeChanges) => {
                    SetVersionCurrent(properties, gltf);
                    SetModelShouldLoad(properties);
                },null),
                CreateModel((properties, gltf, postRuntimeChanges) => {
                    SetVersionFuture(properties, gltf);
                    SetDescription(properties, "Light object added at root");
                    SetModelShouldLoad(properties);
                },(glTFLoader.Schema.Gltf schemaGltf) => { return SetPostRuntimeAtRoot(schemaGltf);
                }),
                CreateModel((properties, gltf, postRuntimeChanges) => {
                    SetVersionFuture(properties, gltf);
                    SetDescription(properties, "Light property added to node object");
                    SetModelShouldLoad(properties);
                },(glTFLoader.Schema.Gltf schemaGltf) => { return SetPostRuntimeInProperty(schemaGltf);
                }),
                CreateModel((properties, gltf, postRuntimeChanges) => {
                    SetVersionFuture(properties, gltf);
                    SetDescription(properties, "Alpha mode updated with a new enum value, and a fallback value");
                    SetModelShouldLoad(properties);
                },  (glTFLoader.Schema.Gltf schemaGltf) => { return SetPostRuntimeWithFallback(schemaGltf);
                }),
                CreateModel((properties, gltf, postRuntimeChanges) => {
                    SetMinVersion(properties, gltf);
                    SetVersionFuture(properties, gltf);
                    SetDescription(properties, "Requires a specific version or higher");
                    SetModelShouldLoad(properties, "Only in version 2.1 or higher");
                },null),
                CreateModel((properties, gltf, postRuntimeChanges) => {
                    SetVersionCurrent(properties, gltf);
                    SetDescriptionExtensionRequired(properties, gltf);
                    SetModelShouldLoad(properties, ":x:");
                },null),
            };
```